post.js

```diff
@@ -114,25 +100,75 @@ class DistributedPost extends HTMLElement {
 
   async loadAndRenderPost(postUrl) {
     if (!postUrl) {
-      console.error("No post URL provided");
+      this.renderErrorContent("No post URL provided");
       return;
     }
 
     let htmlContent;
-    if (postUrl.startsWith("ipfs://")) {
-      htmlContent = await loadPostFromIpfs(postUrl);
-    } else {
-      htmlContent = await loadPost(postUrl);
+    try {
+      if (postUrl.startsWith("ipfs://")) {
+        htmlContent = await loadPostFromIpfs(postUrl);
+      } else {
+        htmlContent = await loadPost(postUrl);
+      }
+
+      const jsonLdUrl = await parsePostHtml(htmlContent);
+      if (jsonLdUrl) {
+        const jsonLdData = await fetchJsonLd(jsonLdUrl);
+        this.renderPostContent(jsonLdData);
+      } else {
+        this.renderErrorContent("JSON-LD URL not found in the post");
+      }
+    } catch (error) {
+      this.renderErrorContent(error.message);
     }
+  }
 
-    const jsonLdUrl = await parsePostHtml(htmlContent);
-    if (jsonLdUrl) {
-      const jsonLdData = await fetchJsonLd(jsonLdUrl);
-      this.innerHTML = renderPost(jsonLdData);
-    } else {
-      console.error("JSON-LD URL not found in the post");
+  renderPostContent(jsonLdData) {
+    // Clear existing content
+    this.innerHTML = "";
+
+    // Create elements for each field
+    if (jsonLdData.attributedTo) {
+      const actorInfo = document.createElement("actor-info");
+      actorInfo.setAttribute("url", jsonLdData.attributedTo);
+      this.appendChild(actorInfo);
+    }
+
+    this.appendField("Summary", jsonLdData.summary);
+    this.appendField("Published", jsonLdData.published);
+    this.appendField("Author", jsonLdData.attributedTo);
+    this.appendField("Content", jsonLdData.content);
+
+    if (jsonLdData.sensitive) {
+      const details = document.createElement("details");
+      const summary = document.createElement("summary");
+      summary.textContent = "Sensitive Content (click to view)";
+      details.appendChild(summary);
+      const content = document.createElement("p");
+      content.textContent = jsonLdData.sensitive;
+      details.appendChild(content);
+      this.appendChild(details);
     }
   }
+
+  appendField(label, value) {
+    if (value) {
+      const p = document.createElement("p");
+      p.innerHTML = `<strong>${label}:</strong> ${value}`;
```

Comment:
Can't use `innerHTML` because it causes XSS vulnerabilities. Strongly advise using HTML templates here instead. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template